src/Server/BaseServer.ts

```diff
@@ -247,45 +256,41 @@ export abstract class BaseServer implements IServer {
   }
 
   /**
-   * Write to a script file
-   * Overwrites existing files. Creates new files if the script does not exist.
+   * Write to a Script or TextFile
+   *    * Overwrites existing files. Creates new files if the script does not exist.
    */
-  writeToScriptFile(filename: ScriptFilePath, code: string): writeResult {
```

Comment:
The two functions were separate for type-checking reasons. I'm leery of just mashing them together. I think you can accomplish the same code-sharing goals by having a private impl that takes care of the shared work.

Reply:
In that case wouldn't it make sense to add the functions back in but have them call writeContentFile? The actual implemention of these functions was identical beside the different variable names so I dont really see a reason not to combine them.
Both textfile and script file inherit the content file interface which should be the interface used to manage the content of these files.

Reply:
It's just a bit inefficient to combine these. If we already know we're dealing with a TextFilePath, then we can directly use writeToTextFile and we're not having to call hasTextExtension again to decide what type of file to write. Same with if we already know it's a ScriptFilePath, doing an extra hasTextExtension check is not helpful.

If the code we're in doesn't already know which type of filepath we're dealing with, that would be when it makes sense to run another hasTextExtension check to determine which type of file to write.

Reply:
This part is still unresolved. I don't think you even need this change anymore for what you're doing?

Reply:
Models are also created for text files, so when a text file is open in the editor and written to it should still update. That's why I kept in the unification of those methods, its removes a lot of imo uneeded code duplication